### PR TITLE
Camera functionality

### PR DESCRIPTION
**Pull request by olivia-livs**
Support for camera arrow key movement, and other adjustments. Looks prettier in the Godot editor.

### CODE REVIEW

**Comment by olivia-livs** [comment]
It says 2 commits but all the changes are from 2dba5bd

**Comment by olivia-livs** [comment]
also AGPLv3 license if wanted.

**Comment by TorchedSammy** [comment]
remove the license for now since there's still a contributor who hasn't commented about the license

could you make it work via right click? haven't actually looked at the code cuz I'm on  a phone

**Comment by olivia-livs** [comment]
they can revert the license commit if wanted. i was trying to make it convenient. do you mean rotate the camera with a right click?

**Comment by olivia-livs** [comment]
ohh wait i get what you mean

**Comment by TorchedSammy** [comment]
> they can revert the license commit if wanted. i was trying to make it convenient

if the pr is for camera movement then keep it to that thanks. it hasnt been decided so 

> i do that will my commits be approved?

after some code checking

**Comment by olivia-livs** [comment]
Ok i removed the license commit, sorry for being annoying about it. Alright thank you :smile:

**Comment by olivia-livs** [comment]
I did it! It works with right click. Took me a lot of trial and error :sob:

File: CameraAnchor.gd [hunk 1]
```diff
@@ -1,34 +1,66 @@
...
+var viewport = 0
+var margin_w = 0
+var margin_h = 0
+var move = Vector2(0, 0)
+var mouse_right_click_origin = Vector2(0, 0)
+var mouse_position = Vector2(0, 0)
+
+#Should move one screen width every 4 seconds
+#var move_speed = viewport.size.x / 4
+var move_speed = 16834 # just manually setting it idk
```
Comment (TorchedSammy) [inline CameraAnchor.gd:13]: is there any reason that you have it set manually?

File: CameraAnchor.gd [hunk 2]
```diff
@@ -1,34 +1,66 @@
...
+	
+	if Input.is_action_pressed("camera_right_click"):
+		mouse_position = get_local_mouse_position()
+		move = mouse_position - mouse_right_click_origin
+		
+		self.velocity = move.normalized() * move_speed * move.length()/256
 		
-		self.velocity = Vector2(move_x * move_vel, move_y * move_vel)
+		move.x = 0
+		move.y = 0
```
Comment (TorchedSammy) [inline CameraAnchor.gd:63]: cant all of this be done in the mouse event motion check? by that you have a toggle for the edge scrolling, but if u dont want to do all that ignore it

**Review by TorchedSammy — CHANGES_REQUESTED** [review]

File: CameraAnchor.gd (hunk 1 above)
Comment (olivia-livs) [reply CameraAnchor.gd:13]: I couldn't get `viewport.size.x / 4` to work.

File: CameraAnchor.gd (hunk 2 above)
Comment (olivia-livs) [reply CameraAnchor.gd:63]: No because `func right_click_movement()` needs to run every frame.

Comment (olivia-livs) [reply CameraAnchor.gd:63]: told you i did a lot of trial and error :sob:

Comment (TorchedSammy) [reply CameraAnchor.gd:63]: hm. i guess ill leave it there then, can you re-request a review?

**Review by TorchedSammy — APPROVED** [review]

**Comment by olivia-livs** [comment]
<3

**Review by AdrienJaguenet — APPROVED** [review]

**Pull request merged**